src/client/src/tscParseLineProtocol.c

```diff
@@ -2077,6 +2078,11 @@ static int32_t parseSmlValue(TAOS_SML_KV *pKV, const char **idx,
     val_rqoute
   } val_state;
 
+  value = malloc(bufSize);
```

Comment:
Use a preallocated buffer in heap instead of a post-allocated one, in order to be able  to construct the result incrementally during processing its syntax.

Reply:
Although it is less efficient to allocate a buffer in heap memory to process short strings than one in stack memory each time, the old behavior of code is preserved for simplicity since the taosAdapter as a stateless independent app which can be scaled horizontally easily on demand.